core/lib/mno_enterprise/app_manager.rb

```diff
@@ -0,0 +1,33 @@
+module MnoEnterprise
+  # Abstract the app management logic
+  #
+  class AppManager
```

Comment:
I would rather be in favour of `SystemManager` to not create any confusion with the `App` model.